sdks/java/harness/src/main/java/org/apache/beam/fn/harness/MapFnRunner.java

```diff
@@ -103,18 +119,49 @@
     }
   }
 
-  private final ThrowingFunction<InputT, OutputT> mapFunction;
+  private final ThrowingFunction<WindowedValue<InputT>, WindowedValue<OutputT>> mapFunction;
   private final FnDataReceiver<WindowedValue<OutputT>> consumer;
 
   MapFnRunner(
-      ThrowingFunction<InputT, OutputT> mapFunction,
+      ThrowingFunction<WindowedValue<InputT>, WindowedValue<OutputT>> mapFunction,
       FnDataReceiver<WindowedValue<OutputT>> consumer) {
     this.mapFunction = mapFunction;
     this.consumer = consumer;
   }
 
   public void map(WindowedValue<InputT> element) throws Exception {
```

Comment:
Instead of creating a wrapper that gets delegated through, create two different inner static classes which just implement the behavior directly.

This would allow us to not explode the windows for a function which doesn't care about the windows and would keep the implementation simple since each individual runner will be trivial and direct.

Reply:
Done. PTAL.